core/GlobalState.cc

```diff
@@ -110,6 +110,16 @@ struct MethodBuilder {
         return *this;
     }
 
+    MethodBuilder &resultTypeUntyped() {
+        method.data(gs)->resultType = Types::untyped(gs, method);
+        return *this;
+    }
+
+    MethodBuilder &resultType(TypePtr &&type) {
+        method.data(gs)->resultType = type;
+        return *this;
+    }
```

Comment:
Could these be done as `buildWithUntypedResult` and `buildWithResult(TypePtr &&type)` and thus replace `build()`?

Reply:
Seems reasonable to me!

Reply:
I couldn't remove `build` completely, but I did make these two functions return the `MethodRef` instead.